### PR TITLE
Feature/db

### PR DESCRIPTION
**Pull request by rcotti-jc**
- Added documents array in Category to store the types of documents required for each category;
it stores objects having a name (String) and a required (boolean)
- Freelance now has a availability array, in which Day objects need to be stored
IMPORTANT: front-end need to include the model where necessary!
added a reply field in review, of max length 1000 characters
- Added a Day Js constructor

### CODE REVIEW

File: models/Category.js [hunk 1]
```diff
@@ -15,14 +15,29 @@ require ('./Freelance');
...
+		//id, certificate diploma, letters
   },
   {
     versionKey: false,
   }
 );
 
 Category.pre('save', function(next) {
+
+	if(this.documents !== null){
```
Comment (larabr) [inline models/Category.js:28]: indentation seems weird but might be github

File: public/js/Day.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	}
+
+	return a;
+}
+
+function sortBegin(day1, day2){
+	return day1.begin < day2.begin;
+}
+
+console.log('x')
```
Comment (larabr) [inline public/js/Day.js]: remove this pls

File: models/Freelance.js [hunk 3]
```diff
@@ -46,6 +46,7 @@ const Freelance = exports.Freelance = new mongoose.Schema({
 		avgScore 			: { type: Number, default: 0 },
 		reviews				: [{ type: ObjectID, ref: "Review", default: [] }],
 		tags					: [{ type: ObjectID, ref: "Tag", default: [] }],
+		availability	: [ {type:Object, default:[] }],
```
Comment (larabr) [inline models/Freelance.js]: space before last ]

**Review by larabr — APPROVED** [review]

File: public/js/Day.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+function Day(begin, end, location, day){
+
+	if(!begin || !end || !location){
+		console.log(69+' a field in a Day is undefined');
```
Comment (aronfiechter) [inline public/js/Day.js:4]: 69?

**Pull request merged**